notifications/management/commands/send_notifications.py

```diff
@@ -274,288 +277,141 @@ def find_bill_search_updates(self, search_params):
-
-        update_groups = []
-
-        outer_grouper = lambda x: x[0]['slug']
-        person_updates = sorted(person_updates, key=outer_grouper)
-
-        inner_grouper = lambda x: x[1]['update_type']
-
-        for slug, group in itertools.groupby(person_updates, key=outer_grouper):
-            bill_group = {}
-
-            group = sorted(group, key=inner_grouper)
-
-            for update_type, inner_group in itertools.groupby(group, key=inner_grouper):
-                bill_group[update_type] = {}
-
-                for person, bill in inner_group:
-                    bill_group[update_type].update(person)
-
-                    try:
-                        bill_group[update_type]['bills'].append(bill)
-                    except KeyError:
-                        bill_group[update_type]['bills'] = [bill]
-
-            update_groups.append(bill_group)
-
-        return update_groups
-
-    def find_committee_action_updates(self, committee_ids):
-        # New actions taken by a committee on a bill
-
-        new_actions = '''
-            SELECT DISTINCT ON (committee.ocd_id, bill.ocd_id, action.id)
-              committee.name,
-              committee.slug,
-              bill.identifier,
-              bill.slug,
-              bill.description,
-              action.description,
-              action.date,
-              action.order
-            FROM councilmatic_core_organization AS committee
-            JOIN councilmatic_core_action AS action
-              ON committee.ocd_id = action.organization_id
-            JOIN councilmatic_core_bill AS bill
-              ON action.bill_id = bill.ocd_id
-            JOIN new_action AS new
-              ON bill.ocd_id = new.bill_id
-            WHERE committee.ocd_id IN %s
-            ORDER BY committee.ocd_id,
-                     bill.ocd_id,
-                     action.id,
-                     action.order DESC
-        '''
-
-        cursor = connection.cursor()
-        cursor.execute(new_actions, [tuple(committee_ids)])
-
-        committee_updates = []
-
-        outer_grouper = lambda x: x[1]
-        inner_grouper = lambda x: x[3]
-        groups = sorted(cursor, key=outer_grouper)
-
-        for committee_slug, group in itertools.groupby(groups, key=outer_grouper):
-
-            group = sorted(group, key=inner_grouper)
-
-            committee_group = {
-                'name': group[0][0],
-                'slug': group[0][1],
+    def find_person_updates(self, person_ids, minutes=15):
+        # Set a base dictionary providing the output data structure for an update
+        base_update_dct = {
+            'New sponsorships': {
+                'slug': '',
+                'name': '',
                 'bills': []
             }
+        }
+        person_updates = []
+        for person_id in person_ids:
+            new_sponsorships = councilmatic_models.BillSponsorship.objects.filter(
+                bill__created_at__gte=self.get_threshold(minutes),
```

Comment:
are sponsorships ever added some time _after_ a bill is created? if so, i don't think this will capture them, but i have the same uncertainty about how to capture changes in related objects as above.